shell/platform/fuchsia/flutter/fuchsia_external_view_embedder.cc

```diff
@@ -462,58 +506,70 @@ void FuchsiaExternalViewEmbedder::SubmitFrame(
-  FML_DCHECK(scenic_views_.find(view_id) == scenic_views_.end());
-
-  ScenicView new_view = {
-      .opacity_node = scenic::OpacityNodeHACK(session_.get()),
-      .entity_node = scenic::EntityNode(session_.get()),
-      .view_holder = scenic::ViewHolder(
-          session_.get(),
-          scenic::ToViewHolderToken(zx::eventpair((zx_handle_t)view_id)),
-          "Flutter::PlatformView"),
-  };
-
-  new_view.opacity_node.SetLabel("flutter::PlatformView::OpacityMutator");
-  new_view.entity_node.SetLabel("flutter::PlatformView::TransformMutator");
-  new_view.opacity_node.AddChild(new_view.entity_node);
-  new_view.entity_node.Attach(new_view.view_holder);
-  new_view.entity_node.SetTranslation(0.f, 0.f,
-                                      -kScenicZElevationBetweenLayers);
-
-  scenic_views_.emplace(std::make_pair(view_id, std::move(new_view)));
-}
+void FuchsiaNextExternalViewEmbedder::CreateView(
+    int64_t view_id,
+    std::optional<ViewProperties> properties) {
+  raster_task_runner_->PostTask([this, view_id, properties]() {
+    FML_DCHECK(scenic_views_.find(view_id) == scenic_views_.end());
+
+    ScenicView new_view = {
+        .opacity_node = scenic::OpacityNodeHACK(session_.get()),
+        .entity_node = scenic::EntityNode(session_.get()),
+        .view_holder = scenic::ViewHolder(
+            session_.get(),
+            scenic::ToViewHolderToken(zx::eventpair((zx_handle_t)view_id)),
+            "Flutter::PlatformView"),
+    };
+
+    new_view.opacity_node.SetLabel("flutter::PlatformView::OpacityMutator");
+    new_view.entity_node.SetLabel("flutter::PlatformView::TransformMutator");
+    new_view.opacity_node.AddChild(new_view.entity_node);
+    new_view.entity_node.Attach(new_view.view_holder);
+    new_view.entity_node.SetTranslation(0.f, 0.f,
+                                        -kScenicZElevationBetweenLayers);
+    if (properties.has_value()) {
+      new_view.pending_hit_testable = properties->hit_testable;
+      new_view.pending_focusable = properties->focusable;
+    }
 
-void FuchsiaExternalViewEmbedder::DestroyView(int64_t view_id) {
-  size_t erased = scenic_views_.erase(view_id);
-  FML_DCHECK(erased == 1);
+    scenic_views_.emplace(std::make_pair(view_id, std::move(new_view)));
+  });
 }
 
-void FuchsiaExternalViewEmbedder::SetViewProperties(int64_t view_id,
-                                                    bool hit_testable,
-                                                    bool focusable) {
-  auto found = scenic_views_.find(view_id);
-  FML_DCHECK(found != scenic_views_.end());
-  auto& view_holder = found->second;
+void FuchsiaNextExternalViewEmbedder::DestroyView(int64_t view_id) {
+  raster_task_runner_->PostTask([this, view_id]() {
+    size_t erased = scenic_views_.erase(view_id);
+    FML_DCHECK(erased == 1);
+  });
+}
 
-  view_holder.pending_hit_testable = hit_testable;
-  view_holder.pending_focusable = focusable;
+void FuchsiaNextExternalViewEmbedder::SetViewProperties(
```

Comment:
Is a call to SetViewProperties ensured to never follow a call to DestroyView? Is it guaranteed because it's tied to the Dart object lifecycle?

Reply:
Nothing guarantees this ATM